src/Dict.h

```diff
@@ -629,7 +691,10 @@ template <typename T> class Dictionary
 		ASSERT(num_entries >= 0);
 		// e is about to be invalid. remove it from all references.
 		if ( order )
-			order->erase(std::remove(order->begin(), order->end(), entry), order->end());
+			{
+			auto hk = entry.GetHashKey();
```

Comment:
This seems unnecessarily expensive, can we do this without copying the key into a new `unique_ptr`?

Reply:
I changed it to be a simple for-loop looking for the key data. Alternatively I could make it a remove_if and a lambda, but it results in the same code in the end.

Reply:
After running into a memory leak with the last change (because move operators are overwriting things as the element is removed), I'm kinda wondering if I need to store the whole key in the order vector after all. Is the hash generated of the key guaranteed to be unique to the point that I could just store the hash value? Then I could add another lookup method to Dictionary (maybe make it protected) that could take the hash value and do the lookup from that?

Reply:
No, (very) strictly speaking the hash won't be sufficient. But now I'm confused, I thought you are already storing the full key in the ordering vector. Need to take another look later.

Reply:
> But now I'm confused, I thought you are already storing the full key in the ordering vector. Need to take another look later.

It is. I was just pondering whether it'd be possible to not have to copy the whole key over there. Not a big deal.

Reply:
Got it, misread. So one idea: could the ordering vector just store a raw pointer to the data from the actual entry? As long as we can manage life-times consistently, that should still be safe.

Reply:
There was a reason I didn't do that, but it's currently escaping me. I want to say it was something about the way we store the HashKey objects in the DictEntry, and the pointers getting screwed up when elements had to move around the Dictionary during remaps.

Reply:
Makes sense.